aep/general/0124/aep.md.j2

```diff
@@ -78,8 +77,54 @@ recommended if additional metadata is required in the relationship, or if the
 restrictions around the use of a repeated field preclude the use of that
 approach.
 
+### Embedded resources
+
+Resource references as described use string references rather than embedding
+one resource inside another because embedding resources can lead to a number of
+issues. Retrieving resources with arbitrarily deep nesting may require multiple
+serial internal RPCs, leading to complex service dependencies latency and
+reliability issues.
+
+However, sometimes dereferencing resource references is useful, or even
+necessary. For example, an API may wish to allow filtering of a resource based
```

Comment:
I feel like the filtering use case could some exploration - is the API design correct if there's a common use case that requires filtering down to a field on a related resource but not the primary resource? what type or parameter would that be?

Reply:
I gave an example live, but for posterity, imagine an InventoryItem resource representing a user's inventory.  InventoryItem has a `oneof` of resource references to other resource types, including - say - AvatarClothing.  And say further that avatar clothing items may be hats, jackets, pants, etc.

A query to express "show me the hats in my inventory" needs to filter on the field of AvatarClothing indicating whether it's a hat vs. a jacket.

Reply:
I've updated the text a bit to have a similar (but book-based) example.